docs/source/upgrade/index.rst

```diff
@@ -145,47 +145,70 @@ We point a browser to http://localhost:4000, and follow the steps:
     #. Exercise the *CoinProposal_Accept* choice on both proposal contracts.
 
 
-Building and deploying version-2
-================================
+Building and deploying coin-2.0.0
+=================================
 
-Now we setup the project for the improved coins containing the *amount* field. This project contains the templates for ``CoinWithAmount``, ``UpgradeCoinProposal`` and ``UpgradeCoinAgreement``.
+Now we setup the project for the improved coins containing the *amount* field. This project contains only the ``CoinWithAmount`` template. The upgrade templates are in a third ``coin-upgrade`` package. While it would be possible to include the upgrade templates in the same package, this means that the package containing the new ``CoinWithAmount`` template depends on the previous version. With the approach taken here of keeping the upgrade templates in a separate package, the ``coin-1.0.0`` package is no longer needed once we have upgraded all coins.
 
-It's worth stressing here that we must setup a new project for version-2. We cannot just add the new definitions to the original project, rebuild and re-deploy. This is because the cryptographically computed package-id would change, and would not match the package-id of version-1 coin-contracts which are live of the ledger.
+It's worth stressing here that extensions always need to go into separate packages. We cannot just add the new definitions to the original project, rebuild and re-deploy. This is because the cryptographically computed package identifier would change, and would not match the package identifier of the original ``Coin`` contracts from ``coin-1.0.0`` which are live of the ledger.
```

Comment:
fix small existing typo. It should be "live on the ledger"

Reply:
Good catch, fixed! Thank you!